sdk/src/Services/DynamoDBv2/Custom/DataModel/ContextInternal.cs

```diff
@@ -196,6 +202,49 @@ internal Table GetUnconfiguredTable(string tableName)
             }
         }
 
+        /// <summary>
+        /// Stores a table in the cache if there is not an existing entry for the given key
+        /// </summary>
+        /// <param name="tableName">Name of the table used as the cache key</param>
+        /// <param name="tableToStore">Table to store if not already present in the cache</param>
+        /// <returns>Table that was either found in the cache or newly stored</returns>
+        internal Table StoreUnconfiguredTable(string tableName, Table tableToStore)
+        {
+            try
+            {
+                _readerWriterLockSlim.EnterReadLock();
+
+                // Check to see if the cache was filled since this was invoked
+                if (tablesMap.TryGetValue(tableName, out Table table))
+                {
+                    return table;
+                }
+            }
+            finally
+            {
+                if (_readerWriterLockSlim.IsReadLockHeld)
+                {
+                    _readerWriterLockSlim.ExitReadLock();
+                }
+            }
+
+            try
+            {
+                _readerWriterLockSlim.EnterWriteLock();
+
+                tablesMap.Add(tableName, tableToStore);
```

Comment:
Now that you have the write lock you need to double check that another thread didn't get the lock before you and add the table to `tablesMap`.

Reply:
Done